es6/README.md

```diff
@@ -51,6 +51,35 @@ globals:
 The boolean value indicates whether the global is to be treated as
 read-only (`false`) or read-write (`true`).
 
+## Naming
+
+Classes and constructor functions should be TitleCased
+
+```javascript
+// GOOD
+class DatePicker { }
+
+var LinkLabel = function() { }
+
+// BAD
+class linkLabel { }
+```
+
+All other symbols (methods, variables, even constants) should be camelCased!
+
+```javascript
+// GOOD
+const defaultName = 'Element'
+
+class MyClass {
+    const name = 'Button'
+    render() { ... }
+}
+
+// BAD - ES6 has `const` now. The compiler/interpreter will warn you if you re-assign a const.
+const DEFAULT_NAME = 'Element'
```

Comment:
The `const` keyword is very useful for indicating things, but it doesn't replace `ALL_CAPS_NAMING` entirely. We get errors for re-assigning a const primitive, but it's different for a composite/mutable value such as an array or object.

Basically, if I see, at the module level,
```js
const stateCallbacks = {
    pending: sendPendingResponse,
    completed: sendCompletedResponse,
    archived: sendArchivedResponse
}
```
then I'd expect that reassigning the members of it would be an intended thing, while if we have 
```js
const STATE_CALLBACKS = {
    pending: sendPendingResponse,
    completed: sendCompletedResponse,
    archived: sendArchivedResponse
}
```
then I'll expect that the contents are intended not to change.